src/resources/js/formdirectives.js

```diff
@@ -0,0 +1,2206 @@
+                }
+            };
+
+            $timeout(function () {
+                $scope.init();
+            });
+        }],
+
+        template: function () {
+            return '<input id="{{id}}" ng-model="model" type="password" class="form-control" autocomplete="{{autocomplete}}" inputmode="{{inputmode}}" />';
+        }
+    }
+});
+
+/**
+ * <zaa-radio model="model" options="[{label:'foo', value: 'bar'}, {...}]"></zaa-radio>
+ */
+zaa.directive("zaaRadio", function () {
+    return {
+        restrict: "E",
+        scope: {
+            "model": "=",
+            "options": "=",
+            "label": "@label",
+            "i18n": "@i18n",
+            "id": "@fieldid",
+            "initvalue": "@initvalue"
+        },
+        controller: ['$scope', '$timeout', function ($scope, $timeout) {
+            $scope.setModelValue = function (value) {
+                $scope.model = value;
+            };
+
+            $scope.init = function () {
+                if ($scope.model === undefined || $scope.model === null) {
+                    $scope.model = typeCastValue($scope.initvalue);
+                }
+            };
+            $timeout(function () {
+                $scope.init();
+            });
+        }],
+        template: function () {
+            return '' +
+                '<div class="form-group form-side-by-side" ng-class="{\'input--hide-label\': i18n}">' +
+                    '<div class="form-side form-side-label">' +
+                        '<label for="{{id}}">{{label}}</label>' +
+                    '</div>' +
+                    '<div class="form-side">' +
+                        '<div ng-repeat="(key, item) in options" class="form-check">' +
+                            '<input value="{{item.value}}" type="radio" ng-click="setModelValue(item.value)" ng-checked="item.value == model" name="{{id}}_{{key}}" class="form-check-input" id="{{id}}_{{key}}">' +
+                            '<label class="form-check-label" for="{{id}}_{{key}}">' +
+                                '{{item.label}}' +
+                            '</label>' +
+                        '</div>' +
+                    '</div>' +
+                '</div>';
+        }
+    };
+});
+
+/**
+ *
+ * Usage Example:
+ *
+ * ```js
+ * <zaa-select model="data.module_name" label="<?= Module::t('view_index_module_select'); ?>" options="modules" />
+ * ```
+ *
+ * If an initvalue is provided, you can not reset the model to null.
+ *
+ * Options value definition:
+ *
+ * ```js
+ * options=[{"value":123,"label":123-Label}, {"value":abc,"label":ABC-Label}]
+ * ```
+ *
+ * In order to change the value and label keys which should be used to take the value and label keys within the given array use:
+ *
+ * ```js
+ * <zaa-select model="create.fromVersionPageId" label="My Label" options="typeData" optionslabel="version_alias" optionsvalue="id"></zaa-select>
+ * ```
+ */
+zaa.directive("zaaSelect", function () {
```

Comment:
So `zaaSelect` would be in formdirectives.js, while `luyaSelect` would be in directives.js. But yes in generall feel free to reanage things and also for sure improve docs and typos, thank you very much!

In general i think we both understand correctly that

"zaaXyz" always includes "luyaXyz".  Agree? So `zaaSelect` will include `<luya-select />` in the template.

Reply:
> So zaaSelect would be in formdirectives.js, while luyaSelect would be in directives.js?
No, both of them will be in formdirectives.js, the second one is right after the first one.

> "zaaXyz" always includes "luyaXyz". Agree? 
Yes. Most of the logic will be in `<luyaXyz>`, while `<zaaXyz>` will be a "form wrapper" the form field.

Please keep in mind that I will strive to specify an `id` attribute only for the HTML input element, so the corresponding attributes for `<luyaXyz>` and `<zaaXyz>` will be `fieldid`. And the `model` attributes will be different for above mentioned directives: `model` for `<zaaXyz>` and `ng-model` for `<luyaXyz>` as it inroduced already in `zaaSelect` and `luyaSelect`.

And the last note concerning compiled javascript. Unglue (and Babel) when compiling javascript uses LF as line ending. My Git autocrlf setting is on, so Git changes the line endings when pushing, so the compliled files are changed too. I hope it's not a problem since you can compile them again just before or after merging.

Reply:
Yes i totally agree with your summary and also its not a problem, i can fix the merge conflict before merging and re compile the files afterwards. I will set the status of the PR back to draft, so you can let me know when things are ready. thanks!